### PR TITLE
add support password for scout redis config

### PR DESCRIPTION
**Pull request by dshvedchenko**
Add support for redis instances required authorization

### CODE REVIEW

**Comment by dshvedchenko** [comment]
Redis protected, scout conf without password:
```
=== RUN   TestWorker_Push
    require.go:794: 
                Error Trace:    worker_client_test.go:79
                Error:          Received unexpected error:
                                NOAUTH Authentication required.
                Test:           TestWorker_Push
--- FAIL: TestWorker_Push (0.01s)
FAIL
exit status 1
FAIL    github.com/enova/scout  0.316s
```
scout test provided with password
```
=== RUN   TestTopicName
--- PASS: TestTopicName (0.00s)
=== RUN   TestWorker_Init
--- PASS: TestWorker_Init (0.00s)
=== RUN   TestWorker_Push
--- PASS: TestWorker_Push (0.03s)
PASS
```

**Comment by dshvedchenko** [comment]
I tried to build with docker, but could not

File: config_test.go [hunk 1]
```diff
@@ -85,3 +86,20 @@ func (c *ConfigTestSuite) TestConfig_Sparse() {
...
+
+// It's ok for stuff to be missing, we'll check that elsewhere
+func (c *ConfigTestSuite) TestConfig_RedisPass() {
+	c.WriteTemp(redisAuthConfig)
+	config, err := ReadConfig(c.tempfile.Name())
+	c.assert.NoError(err)
+
+	c.assert.Equal(config.Redis.Password, "jsdf3-402341234")
+	c.assert.Equal(config.Redis.Host, "localhost:9000")
+}
```
Comment (mokpro) [inline config_test.go:105]: add a newline at the end of the file

File: worker_client.go [hunk 2]
```diff
@@ -27,13 +27,26 @@ func NewRedisWorkerClient(redis RedisConfig) (WorkerClient, error) {
...
+	} else {
+		workers.Configure(map[string]string{
+			"server":    redis.Host,
+			"database":  "0",
+			"pool":      "20",
+			"process":   "1",
+			"namespace": redis.Namespace,
+			"password": redis.Password,
+		})	
+	}
```
Comment (mokpro) [inline worker_client.go:48]:
```suggestion
workerConfig := map[string]string{
			"server":    redis.Host,
			"database":  "0",
			"pool":      "20",
			"process":   "1",
			"namespace": redis.Namespace,
		}

if redis.Password != "" {
  workerConfig["password"] = redis.Password
}

workers.Confugure(workerConfig)
```

**Review by mokpro — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -43,6 +43,8 @@ redis:
   host: "localhost:9000"
   namespace: "test"
   queue: "background"
+  #
```
Comment (lladewig) [inline README.md:46]: This `#` seems unnecessary

**Review by lladewig — CHANGES_REQUESTED** [review]
also please run `gofmt`

File: README.md (hunk 3 above)
Comment (vbhat) [inline README.md:46]:
```suggestion
  
```

File: worker_client.go (hunk 2 above)
Comment (vbhat) [reply worker_client.go:48]: Please make this change, but don't auto-commit. There's a typo in `workers.Configure(workerConfig)`

**Review by mokpro — APPROVED** [review]
this looks good to me 👍 

Waiting for CI fix in #14. I can merge it after that

**Comment by dshvedchenko** [comment]
ok , we then rebase it on #14

**Pull request closed without merging**